File: SonicXBackendC/SonicXBackendC/Server/Email/VerifyEmailHandler.cs
using System.IO;
using Newtonsoft.Json;
using System.Net;
using SonicXBackendC.UserManagement;
using System.Text;
using System;

namespace SonicXBackendC.Server.Email
{
    public class VerifyEmailHandler
    {
        public void Handle(HttpListenerContext context)
        {
            if (context.Request.HttpMethod == "POST")
            {
                using (var reader = new StreamReader(context.Request.InputStream))
                {
                    var requestBody = reader.ReadToEnd();
                    var requestData = JsonConvert.DeserializeObject<RequestData>(requestBody);

                    Console.WriteLine("Gelen veri: " + requestData);

                    var userService = UserService.GetInstance();
                    bool success = userService.VerifyEmail(requestData.Email, requestData.VerificationCode);

                    var responseJson = JsonConvert.SerializeObject(new { success });

                    context.Response.ContentType = "application/json";
                    context.Response.StatusCode = (int)HttpStatusCode.OK;
                    var responseBytes = Encoding.UTF8.GetBytes(responseJson);
                    context.Response.OutputStream.Write(responseBytes, 0, responseBytes.Length);
                }
            }
            else
            {
                context.Response.StatusCode = (int)HttpStatusCode.MethodNotAllowed;
            }
        }
    }
}

File: SonicXBackned/src/main/java/org/example/Server/Email/VerifyEmailHandler.java
package org.example.Server.Email;


import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.example.Server.RequestData;
import org.example.UserManagment.UserService;

import java.io.IOException;
import java.io.OutputStream;

public class VerifyEmailHandler implements HttpHandler {
    @Override
    public void handle(HttpExchange exchange) throws IOException {
        if ("POST".equals(exchange.getRequestMethod())) {
            ObjectMapper objectMapper = new ObjectMapper();
            RequestData requestData = objectMapper.readValue(exchange.getRequestBody(), RequestData.class);

            System.out.println("Gelen veri: " + requestData);

            UserService userService = UserService.getInstance();
            boolean success = userService.verifyEmail(requestData.getEmail(), requestData.getVerificationCode());

            ObjectNode responseJson = objectMapper.createObjectNode();
            responseJson.put("success", success);

            byte[] response = responseJson.toString().getBytes();
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, response.length);
            OutputStream os = exchange.getResponseBody();
            os.write(response);
            os.close();
        } else {
            exchange.sendResponseHeaders(405, -1);
        }
    }
}

